tracim/views/contents_api/comment_controller.py

```diff
@@ -0,0 +1,166 @@
+# coding=utf-8
+import transaction
+from pyramid.config import Configurator
+
+try:  # Python 3.5+
+    from http import HTTPStatus
+except ImportError:
+    from http import client as HTTPStatus
+
+from tracim import TracimRequest
+from tracim.extensions import hapic
+from tracim.lib.core.content import ContentApi
+from tracim.lib.core.workspace import WorkspaceApi
+from tracim.lib.utils.authorization import require_workspace_role
+from tracim.lib.utils.authorization import require_comment_ownership_or_role
+from tracim.views.controllers import Controller
+from tracim.views.core_api.schemas import CommentSchema
+from tracim.views.core_api.schemas import CommentsPathSchema
+from tracim.views.core_api.schemas import SetCommentSchema
+from tracim.views.core_api.schemas import WorkspaceAndContentIdPathSchema
+from tracim.views.core_api.schemas import NoContentSchema
+from tracim.exceptions import WorkspaceNotFound
+from tracim.exceptions import InsufficientUserWorkspaceRole
+from tracim.exceptions import NotAuthenticated
+from tracim.exceptions import AuthenticationFailed
+from tracim.models.contents import ContentTypeLegacy as ContentType
+from tracim.models.revision_protection import new_revision
+from tracim.models.data import UserRoleInWorkspace
+
+COMMENT_ENDPOINTS_TAG = 'Comments'
+
+
+class CommentController(Controller):
+
+    @hapic.with_api_doc(tags=[COMMENT_ENDPOINTS_TAG])
+    @hapic.handle_exception(NotAuthenticated, HTTPStatus.UNAUTHORIZED)
+    @hapic.handle_exception(InsufficientUserWorkspaceRole, HTTPStatus.FORBIDDEN)
```

Comment:
`InsufficientUserWorkspaceRole` is probably to rename to `InsufficientUserRoleInWorkspace`. The reason for that is "it's understandable as naturally read"